### PR TITLE
list_box

### PR DESCRIPTION
**Pull request by gintama91**
Shoes::list_box

### CODE REVIEW

**Comment by noahgibbs** [comment]
This is failing because of the linter, mostly on whitespace errors. @schwad - how do we feel about that?

**Comment by Schwad** [comment]
@noahgibbs I feel that having the RubyLSP enabled when coding would auto format these on save. 

But I appreciate that as an OSS project we can't expect folks to all have the same setup and I want things to be slick for people.

Do you know what our options or common practices would be around having autocorrectable offences cleaned up by CI?

I'm visualizing maybe we don't fail unless there are uncorrectable offenses, then maybe it lints on another commit during merge?

Like I wonder if there's a GitHub action for that.

For example in this case, 'bundle exec rubocop . --autocorrect-all' would've fixed these whitespaces and it wouldn't have needed to fail 🤔

**Comment by Schwad** [comment]
Like, I'm for aggressively tamping down offences that get in our creative way BUT I do find whitespace cleanup of stylistic use when coding. The risk merging here is if I touch the file and save it's likely to later do a bunch of cleanup changes that could confuse the reviewer 

Open to your thoughts and ideas!

**Comment by noahgibbs** [comment]
As far as CI, your choices are:

* GitHub Action to clean it up by making an additional commit -- which will at least occasionally break something
* Git Hooks to do a cleanup before commit, but contributors will have to install them

The second one makes having a CI step that fails more tolerable by moving the failure earlier in the process. It's not friendly, but it's arguably friendlier than having CI fail and having to dig.

The first one would *strongly* suggest us ditching "Rubocop defaults minus the ones we hate most" in favour of "a short curated list of rules selected to autocorrect but never cause build breaks." We might be better off switching to something like rubyfmt that is designed to *never* change the semantics.

**Comment by noahgibbs** [comment]
Also, any kind of autocorrect is going to occasionally cause weirdness. For instance, I know *I* get constant Rubocop failures on any kind of actual Shoes code, which is why we exclude the entire examples directory from it. Even if we switched to rubyfmt, we'd want to protect Shoes code from it, or none of our Shoes examples are going to look like the originals.

We can adopt an assertive style guide, roughly analogous to a newspaper having a strong style guide for English, but it's going to look weird when you put anything that is *not* a newspaper article into that format.

**Comment by Schwad** [comment]
Okay, thanks for the context! I'm open to reviewing rubocop generally, I believe SyntaxTree has an autoformatting component support for IDEs so I might look into that. I'm sure it's much less onerous with its rules but could help me with things I'm considered about. 

Like it might not force you out of colonmethodcall, for example.

I'll throw up an issue.

In the short-term it is worth ensuring the IDE setups are working for our contributors manually so they don't have the headwinds from CI (or at least doing lint checks as part of the pre-push ritual locally)

thank you for your thoughts!

**Comment by gintama91** [comment]
this ci tests cancelled(prolly host didn't come online or something that sort) 
before lint commit ci tests were passing so i assume they pass now too or anyone can rerun tests ig?
https://github.com/scarpe-team/scarpe/actions/runs/5101410089/jobs/9170226723

File: examples/legacy/not_checked/simple/form.rb [hunk 1]
```diff
@@ -4,7 +4,9 @@
   background "../static/menu-left.png", :top => 50, :width => 55
   background "../static/menu-right.png", :right => 0, :top => 50, :width => 55
   image "../static/menu-corner1.png", :top => 0, :left => 0
-  image "../static/menu-corner2.png", :right => 0, :top => 0
+  # image "../static/menu-corner2.png", :right => 0, :top => 0
```
Comment (Schwad) [inline examples/legacy/not_checked/simple/form.rb:7]: Looking pretty good! Do we want to resolve the commented code?

Comment (gintama91) [reply examples/legacy/not_checked/simple/form.rb:7]: yeah image dont work with local files as of now ig? tho its not showing any error 

that's the reason i haven't moved that file to working directory yet

Comment (gintama91) [reply examples/legacy/not_checked/simple/form.rb:7]: and ig we need this 

https://github.com/shoes/shoes3/tree/master/static

folder into our branch for many examples ig for images

Comment (gintama91) [reply examples/legacy/not_checked/simple/form.rb:7]: oh nvm it is in docs/static directory

**Pull request merged**